src/main/java/com/snowflake/kafka/connector/internal/streaming/DirectTopicPartitionChannel.java

```diff
@@ -323,14 +328,20 @@ public void insertRecord(SinkRecord kafkaSinkRecord, boolean isFirstRowPerPartit
     if (currentProcessedOffset == NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE
         || kafkaSinkRecord.kafkaOffset() >= currentProcessedOffset + 1) {
       transformAndSend(kafkaSinkRecord);
+      isFirstMismatchedOffset.set(true);
```

Comment:
I'm not a fan of logging something on the first occurrence and later skipping. 
I propose to keep it consistent.

Reply:
ok, lets keep it as info then. warn seems to me too high.

Reply:
Will it be OK that we print this infinite times, especially in the case of expected offset being earlier to current offset?

Personally I think its alright since an action is certainly required but I was just wondering from a performance/cost impact to customers (and iff any of our previous logging decisions were led by that).